### ISSUE TITLE
Add Metrics functions for TensorFlow frontend

### ISSUE THREAD

**Issue by sreegandh804** [unifyai/ivy#1547]
Add [Metrics functions](https://www.tensorflow.org/api\_docs/python/tf/keras/metrics#functions\_2) for TensorFlow frontend:

\_

>Please keep in mind that the proper way to link an issue to this list is to comment "- [ ] #issue\_number" while the issue's title only includes the name of the function you've chosen.

\_

- [ ] binary\_accuracy
- [x] #10193
- [x] #5569
- [x] #3975
- [ ] categorical\_crossentropy
- [x] #9680
- [x] #9681
- [ ] hinge
- [x] #3989
- [ ] #5018
- [x] #5019
- [x] #5360
- [x] #7017
- [x] #4668
- [x] #5020
- [ ] mean\_absolute\_error 
- [x] #4060
- [x] #2978
- [x] #3083
- [ ] #5021
- [x] #5022
- [x] #4023
- [ ] serialize
- [x] #5643
- [ ] sparse\_categorical\_crossentropy
- [ ] sparse\_top\_k\_categorical\_accuracy
- [ ] squared\_hinge
- [ ] top\_k\_categorical\_accuracy

**Comment by siddartha-10** [comment]
#1554

**Comment by sreegandh804** [comment]
Hi @siddartha-10 ,
The above tasks are not open to work on yet. Feel free to checkout the following link to see the open tasks you could work on : https://github.com/unifyai/ivy/blob/bf11369d9f500b969decbc4212467464c2fa4f2f/docs/partial_source/contributing/4_open_tasks.rst#frontend-apis . Let me know if you have any other question 
Thanks!
Sreegandh 😃

**Comment by wammy19** [comment]
#2894

**Comment by ztsv-av** [comment]
#2978

**Comment by tosmartak** [comment]
#2997

**Comment by rahulsnkr** [comment]
- [ ] #3083

**Comment by JLU-Neal** [comment]
#3152

**Comment by JLU-Neal** [comment]
#3271

**Comment by wammy19** [comment]
#3285

**Comment by HasanResul** [comment]
- [ ] #3465

**Comment by AhmedMousa-ag** [comment]
- [ ]#3502

**Comment by AhmedMousa-ag** [comment]
- [ ]#3502

**Comment by mustafa1adel** [comment]
#3551

**Comment by mustafa1adel** [comment]
[] #3551

**Comment by mustafa1adel** [comment]
- [ ] #3551

**Comment by shunleili** [comment]
sparse_categorical_accuracy #3751

**Comment by MutugiD** [comment]
#4999

**Comment by MutugiD** [comment]
- [ ] #4999

**Comment by khimraj** [comment]
- [ ] #5018

**Comment by khimraj** [comment]
- [ ] #5019

**Comment by khimraj** [comment]
- [ ] #5020

**Comment by khimraj** [comment]
- [ ] #5021

**Comment by khimraj** [comment]
- [ ] #5022

**Comment by MutugiD** [comment]
- [ ] #5027

**Comment by jordibardia** [comment]
log_cosh #5360

**Comment by medphysicsliv** [comment]
-[]#5569

**Comment by Adityahulk** [comment]
[]#5643

**Comment by MutugiD** [comment]
top_k_categorical_accuracy #6341

**Comment by khimraj** [comment]
- [ ] #7017

**Comment by khimraj** [comment]
- [ ] #7236

**Comment by JoelPendleton** [comment]
- [ ] #7255

**Comment by JoelPendleton** [comment]
- [ ] #7256

**Comment by guptaru1** [comment]
categorical accuracy #3975

**Comment by jiten597** [comment]
mean_absolute_error #7831

**Comment by jiten597** [comment]
categorical_crossentropy #7833

**Comment by jiten597** [comment]
binary_crossentropy #7835

**Comment by obaidhsn** [comment]
- [ ] #9609

**Comment by AdvancedMicroDev** [comment]
- [ ] #9680

**Comment by AdvancedMicroDev** [comment]
- [ ] #9681

**Comment by yash-garg274** [comment]
#9966

**Comment by david-adewoyin** [comment]
#10193

**Comment by gaurav2255** [comment]
mean_absolute_error #11195

**Comment by Manish06097** [comment]
i want to contribute in  this sub task categorical_crossentropy #1547

**Comment by srihari-sirisipalli** [comment]
- categorical_crossentropy #2023

**Comment by Manish06097** [comment]
#11500

**Comment by ajithpious** [comment]
- [ ] #11690

**Comment by obaidhsn** [comment]
- [ ] #11691

**Comment by obaidhsn** [comment]
- [ ] #11692
